tools/db_stress.cc

```diff
@@ -1814,6 +1849,46 @@ class StressTest {
         } else {
           MultiDelete(thread, write_opts, column_family, key);
         }
+      } else if (delBound <= prob_op && prob_op < delRangeBound) {
+        // OPERATION delete range
+        if (!FLAGS_test_batches_snapshots) {
+          std::vector<std::unique_ptr<MutexLock>> range_locks;
+          range_locks.emplace_back(std::move(l));
+          // delete range does not respect disallowed overwrites. the keys for
+          // which overwrites are disallowed are randomly distributed so it
+          // could be expensive to find a range where each key allows
+          // overwrites.
+          while (rand_key > max_key - FLAGS_range_deletion_width) {
+            range_locks[0].reset();
+            rand_key = thread->rand.Next() % max_key;
+            rand_column_family = thread->rand.Next() % FLAGS_column_families;
+            range_locks[0].reset(new MutexLock(
```

Comment:
why you need to lock the key if `rand_key > max_key - FLAGS_range_deletion_width` does not satisfy?